avocado/plugins/diff.py

```diff
@@ -0,0 +1,427 @@
+
+        if (('all' in args.diff_filter or 'results' in args.diff_filter) and
+           'noresults' not in args.diff_filter):
+
+            results1 = []
+            for test in job1_data['tests']:
+                test_result = '%s: %s\n' % (str(test['url']),
+                                            str(test['status']))
+                results1.append(test_result)
+            results2 = []
+            for test in job2_data['tests']:
+                test_result = '%s: %s\n' % (str(test['url']),
+                                            str(test['status']))
+                results2.append(test_result)
+
+            if str(results1) != str(results2):
+                test_results_header = ['\n',
+                                       'TEST RESULTS\n',
+                                       '============\n']
+                job1_results.extend(test_results_header)
+                job1_results.extend(results1)
+                job2_results.extend(test_results_header)
+                job2_results.extend(results2)
+
+        if (('all' in args.diff_filter or 'config' in args.diff_filter) and
+           'noconfig' not in args.diff_filter):
+
+            config1 = self._get_config(job1_dir)
+            config2 = self._get_config(job2_dir)
+
+            if str(config1) != str(config2):
+                config_header = ['\n',
+                                 'SETTINGS\n',
+                                 '========\n']
+                job1_results.extend(config_header)
+                job1_results.extend(config1)
+                job2_results.extend(config_header)
+                job2_results.extend(config2)
+
+        if (('all' in args.diff_filter or 'sysinfo' in args.diff_filter) and
+           'nosysinfo' not in args.diff_filter):
+
+            sysinfo_pre1 = self._get_sysinfo(job1_dir, 'pre')
+            sysinfo_pre2 = self._get_sysinfo(job2_dir, 'pre')
+
+            if str(sysinfo_pre1) != str(sysinfo_pre2):
+                sysinfo_header_pre = ['\n',
+                                      'SYSINFO PRE\n',
+                                      '===========\n']
+                job1_results.extend(sysinfo_header_pre)
+                job1_results.extend(sysinfo_pre1)
+                job2_results.extend(sysinfo_header_pre)
+                job2_results.extend(sysinfo_pre2)
+
+            sysinfo_post1 = self._get_sysinfo(job1_dir, 'post')
+            sysinfo_post2 = self._get_sysinfo(job2_dir, 'post')
+
+            if str(sysinfo_post1) != str(sysinfo_post2):
+                sysinfo_header_post = ['\n',
+                                       'SYSINFO POST\n',
+                                       '============\n']
+                job1_results.extend(sysinfo_header_post)
+                job1_results.extend(sysinfo_post1)
+                job2_results.extend(sysinfo_header_post)
+                job2_results.extend(sysinfo_post2)
+
+        if getattr(args, 'create_reports', False):
```

Comment:
well you know those options are present, you can use `args.create_reports` directly... (the same on other places) (can be tweaked after the release)